types/sshloglevel.pp

```diff
@@ -0,0 +1,3 @@
+# The possible openssh LogLevel values. Note that these are values allowed by the
+# smart_proxy_remote_execution_ssh gem and are not consistent with the ssh_config(5) man page.
+type foreman_proxy::sshloglevel = enum['debug', 'info', 'error', 'fatal']
```

Comment:
What is the benefit of abstracting these values to a types file versus just defining an enum directly in the class?

Reply:
DRYing the code (which isn't an issue here), readability in that the param declaration is better self documenting, and better documentation in that there are docs for the enum that don't have to be part of the param's doc string.